jbmc/regression/jbmc-strings/ConstantEvaluationStringBufferDelete/Test.java

```diff
@@ -0,0 +1,72 @@
+public class Test
+{
+  public void testSuccess1()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(0, 0);
+    assert sb.toString().equals("abc");
+  }
+
+  public void testSuccess2()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(0, 3);
+    assert sb.toString().equals("");
+  }
+
+  public void testSuccess3()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(0, 4);
+    assert sb.toString().equals("");
+  }
+
+  public void testSuccess4()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(1, 2);
+    assert sb.toString().equals("ac");
+  }
+
+  public void testSuccess5()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(3, 4);
+    assert sb.toString().equals("abc");
+  }
+
+  public void testNoPropagation1(int index)
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(index, 3);
+    assert sb.toString().equals("ac");
+  }
+
+  public void testNoPropagation2(int index)
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(0, index);
+    assert sb.toString().equals("ac");
+  }
+
+  public void testIndexOutOfBounds1()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    sb.delete(-1, 3);
```

Comment:
Normally I'd expect this to throw and therefore the assertion is unreachable; are you checking for a thrown exception here or do we deviate from the real StringBuilder's behaviour?

Reply:
Yes this throws an exception. I'll remove the assertion and check specifically that an exception is thrown.